torch/csrc/lazy/backend/backend_data.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <string.h>
+#include <torch/csrc/lazy/core/shape.h>
+
+namespace torch {
+namespace lazy {
+
+class BackendData {
+ public:
+  struct Info {
+    /**
+     * Used by Lazy Graph Executor to tag info on BackendData objs
+     * */
+    virtual ~Info() {}
+  };
+  /**
+   * Represents (Tensor) data stored on a backend device
+   * in its native format.
+   * */
+  using Handle = int64_t;
+
+  // TODO(whc) change device to lazy::Device? or delete it?
+  BackendData(std::string device, Shape shape)
+      : device_(std::move(device)), shape_(std::move(shape)) {}
+
+  virtual ~BackendData() {}
+
+  const std::string& device() const {
+    return device_;
+  }
+
+  const Shape& shape() const {
```

Comment:
One somewhat related comment. Since LTC decide to keep the `torch::lazy::shape` on the core level, is it a good idea to actually add `torch::lazy::shape` to the Base IR node? This way we don't need to keep the current `getShape(IR)` call. I can also make `XlaNode` to have a `xlaShape()` method and slowly replaced the usage of it to `shape()`.

Reply:
Yea, I think we can add it back to Node base.  Jiewen and you have both been twisting my arm.  :)

I think @Krovatkin had some reservations about it all along; that we don't want shape in the IR simply because we want the IR reusable.  Nick, are you good with this, maybe if we put a comment explaining the shape is useful for upper-bounds inference but shouldn't be relied on in TS backend?  Any other concerns?

Reply:
@wconstab 
I think it does make a lot of sense to provide this utility for other vendors, too. 

> maybe if we put a comment explaining the shape is useful for upper-bounds inference but shouldn't be relied on in TS backend? Any other concerns?

I wonder if we could code-document it by maybe renaming `shape` to `upper_bounds` or something like that? Or would it break too many things downstream and in LTC? @JackCaoG

Reply:
it doesn't matter too much for pt/xla right now as we will keep using `xlashape()` in the short term. The plan is slowly migrate to use `lazy::shape`  from lazy tensor core. I actually think it is better to name it `shape` since in the static mode `upper_bound` is confusing. Plus `Shape` also encasualte the `c10::ScalarType`.